scene/main/node.cpp

```diff
@@ -2487,7 +2487,7 @@ bool Node::has_node_and_resource(const NodePath &p_path) const {
 	Vector<StringName> leftover_path;
 	Node *node = get_node_and_resource(p_path, res, leftover_path, false);
 
-	return (node && res.is_valid());
+	return node;
```

Comment:
When it's alone like this I prefer to make the conversion to bool explicit, so that it doesn't look like a logic mistake, so `return (node != NULL)`.

That being said, doesn't that defeat the purpose of `has_node_and_resource`, at least the one I documented? This would make it behave basically the same as `has_node()`, no?

Reply:
I changed `get_node_and_resource` so it returns `NULL` when there is no such resource.

Reply:
Alright, that makes sense.